webApp/src/main/java/com/infoshare/controller/NeedRequestController.java

```diff
@@ -17,9 +28,30 @@ public NeedRequestController(NeedRequestService needRequestService) {
         this.needRequestService = needRequestService;
     }
 
+    @PostMapping("/submit-new-form")
+    public String submitNeedRequestForm(@Valid @ModelAttribute("needRequestForm") NeedRequestForm needRequestForm, BindingResult br, Model model) {
+        if (br.hasErrors()) {
+            List<TypeOfHelp> typeOfHelp = Arrays.asList(TypeOfHelp.values());
+            model.addAttribute("types", typeOfHelp);
```

Comment:
Przeniósłbym wszystko co tylko się da do klasy NeedRequestService:
```suggestion
            model.addAttribute("types", needRequestService.getAllTypesOfHelp());
```
Można wówczas wykorzystać tę metodę zarówno dla requestu get, jak i post, i dzięki temu uniknąć duplikacji kodu oraz pozbyć się części logiki z kontrolera.